debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -318,7 +318,7 @@ private static String buildCacheKey(Struct value, Struct originalRecordValue) {
         // But since ExtractChangedRecordState generates changed field with `update` or `delete` operation and then eventually copied to the payload with HeaderToValue SMT,
         // the schema in that case will never be updated since cached on the first `read` operation.
         // Using also the operation in the cache key will solve the problem.
-        return value.schema() + originalRecordValue.getString(OPERATION);
+        return value.schema().hashCode() + originalRecordValue.getString(OPERATION);
```

Comment:
Don't you think it would bet better to use value.schema() + value.schema().hashCode() + originalRecordValue.getString(OPERATION) to avoid potential hash collisions with unrelated schemas?

Reply:
I prefer to use topic name instead of same value of `value.schema()`

Reply:
@harveyyue There is still an issue of having routing SMT applied on the source side and sending multiple tables to a single topic. In that case you'll end up with the collisions again. WDYT?

Reply:
@jpechane Yes, you are right, I just want to reduce the probability of hash collisions. However, we should add a static class like `NewRecordValueMetadata` to fix it.